tests/aclorch_ut.cpp

```diff
@@ -0,0 +1,1075 @@
+#include "ut_helper.h"
+
+extern sai_object_id_t gSwitchId;
+
+extern CrmOrch* gCrmOrch;
+extern PortsOrch* gPortsOrch;
+extern RouteOrch* gRouteOrch;
+extern IntfsOrch* gIntfsOrch;
+extern NeighOrch* gNeighOrch;
+
+extern FdbOrch* gFdbOrch;
+extern MirrorOrch* gMirrorOrch;
+extern VRFOrch* gVrfOrch;
+
+extern sai_acl_api_t* sai_acl_api;
+extern sai_switch_api_t* sai_switch_api;
+extern sai_port_api_t* sai_port_api;
+extern sai_vlan_api_t* sai_vlan_api;
+extern sai_bridge_api_t* sai_bridge_api;
+extern sai_route_api_t* sai_route_api;
+
+namespace nsAclOrchTest {
+
+using namespace std;
+
+static size_t consumerAddToSync(Consumer* consumer, const std::deque<KeyOpFieldsValuesTuple>& entries)
+{
+    /* Nothing popped */
+    if (entries.empty()) {
+        return 0;
+    }
+
+    for (auto& entry : entries) {
+        string key = kfvKey(entry);
+        string op = kfvOp(entry);
+
+        /* If a new task comes or if a DEL task comes, we directly put it into getConsumerTable().m_toSync map */
+        if (consumer->m_toSync.find(key) == consumer->m_toSync.end() || op == DEL_COMMAND) {
+            consumer->m_toSync[key] = entry;
+        }
+        /* If an old task is still there, we combine the old task with new task */
+        else {
+            KeyOpFieldsValuesTuple existing_data = consumer->m_toSync[key];
+
+            auto new_values = kfvFieldsValues(entry);
+            auto existing_values = kfvFieldsValues(existing_data);
+
+            for (auto it : new_values) {
+                string field = fvField(it);
+                string value = fvValue(it);
+
+                auto iu = existing_values.begin();
+                while (iu != existing_values.end()) {
+                    string ofield = fvField(*iu);
+                    if (field == ofield)
+                        iu = existing_values.erase(iu);
+                    else
+                        iu++;
+                }
+                existing_values.push_back(FieldValueTuple(field, value));
+            }
+            consumer->m_toSync[key] = KeyOpFieldsValuesTuple(key, op, existing_values);
+        }
+    }
+    return entries.size();
+}
+
+struct AclTestBase : public ::testing::Test {
```

Comment:
>AclTestBase [](start = 7, length = 11)

Name is confusing. Has nothing related Acl.

Reply:
I think the name is no confuse. That is base class of AclTest and AclOrchTest.